### PR TITLE
LB communication with Kuzzle

### PR DESCRIPTION
**Pull request by AnthonySendra**
> :warning: Must be merged along with [this Kuzzle PR](https://github.com/kuzzleio/kuzzle/pull/284)

* Refactor envelope communication between LB and Kuzzle
* Uniformize context/connection

### CODE REVIEW

File: lib/service/Broker.js [hunk 1]
```diff
@@ -93,41 +93,41 @@ Broker.prototype.onError = function (error) {
...
   else if (!server) {
-    this.pendingRequestStore.add(pendingItem);
+    this.brokerRequestStore.add(pendingItem);
   }
   else {
     // The broker delegates the pending to the server
-    this.pendingRequestStore.remove(pendingItem);
+    this.brokerRequestStore.remove(pendingItem);
+    util = require('util');
+    console.log(util.inspect(pendingItem.message, {showHidden: false, depth: null}));
```
Comment (ballinette) [inline lib/service/Broker.js]: debug trace to clean ?

**Comment by ballinette** [comment]
+1, but probably need to cleanup the debug traces that seem to break the linter ;)

File: test/store/clientConnection.test.js [hunk 2]
```diff
@@ -99,37 +99,37 @@ describe('Test: store/ClientConnection', function () {
...
     clientConnection.clientConnections = {
-      [dummyConnectionExist.socketId]: dummyConnectionExist
+      [dummyConnectionExist.id]: dummyConnectionExist
     };
 
     should(clientConnection.get(dummyInvalidConnection)).be.undefined();
   });
 
-  it('method getBySocketId must return an item from clientConnections if it exists', () => {
+  it('method getByid must return an item from clientConnections if it exists', () => {
```
Comment (dbengsch) [inline test/store/clientConnection.test.js]: getByConnectionId instead of getByid

**Comment by dbengsch** [comment]
apart from my nitpicking : :100:

**Comment by j33f** [comment]
+1

**Pull request merged**